api.bs

```diff
@@ -855,6 +864,52 @@ if the user has opted out of collection of diagnostic data.
 * User ability to view the impression store and past report submissions.
 
 
+# HTTP API # {#http-api}
+
+\`<dfn http-header><code>Save-Impression</code></dfn>\` is a
+[=structured header/dictionary|Dictionary Structured Header=]
+set on a response requesting that the user agent invoke the
+<a method for=PrivateAttribution>saveImpression()</a> API.
+
+<pre class=example id=ex-save-impression-header>
+Save-Impression: conversion-site=advertiser.example, histogram-index=2, \
+  filter-data=12, lifetime-days=7
+</pre>
+
+The following keys are defined, corresponding to the members of
+the {{PrivateAttributionImpressionOptions}} dictionary passed to
+<a method for=PrivateAttribution>saveImpression()</a>.
+
+<dl dfn-for=save-impression>
+  <dt><code>conversion-site</code></dt>
+  <dd>
+    Value of <a dict-member for=PrivateAttributionImpressionOptions>conversionSite</a>,
+    a [=structured header/string=]. This key is required,
+    and must be a valid domain name.
+  </dd>
+  <dt><code>histogram-index</code></dt>
+  <dd>
+    Value of <a dict-member for=PrivateAttributionImpressionOptions>histogramIndex</a>,
+    an [=structured header/integer=]. This key is required.
+  </dd>
+  <dt><code>filter-data</code></dt>
+  <dd>
+    Value of <a dict-member for=PrivateAttributionImpressionOptions>filterData</a>.
+    Implementations parsing the header must accept any valid
+    [=structured header/item=] or [=structured header/inner list=].
+    If the filter data is not understood, no impression is saved.
```

Comment:
Are these sentences in conflict?

And what are the implications of the first sentence for how [filter-data matching](https://patcg.github.io/ppa-api/#logic-matching) works?

Reply:
Yeah, it doesn't make a lot of sense as written. I was trying to allow for extensibility of the filter data in the future, but not saving an impression sounds a lot like rejecting the filter data.

Reply:
I think that we can probably constrain the filter data in both the API and here to avoid the need to allow for problems.  There's a future-proofing argument to be had, but perhaps the way to deal with that is that we add a new dictionary memory (API) or key (HTTP) if we want to change what can be stored with an impression.  Right now, [`filterData`]() is an `unsigned long`, so maybe the answer is to define this as an integer, which is set to zero if the key is absent or the key contains a value that does not fit into the value space of `unsigned long`.

Reply:
+1 to making this an integer for now (we can always accept other types backwards-compatibly in the future), but it's unclear to me if the absence of the field should mean "match any filter data value" or if, as you say, it should be set to `0`, which would make it match only `0`.

CC #57.